web/js/MapUI/Components/UpdateProjection/UpdateProjection.js

```diff
@@ -0,0 +1,399 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import OlLayerGroup from 'ol/layer/Group';
+import {
+  each as lodashEach,
+  get as lodashGet,
+} from 'lodash';
+import {
+  fitToLeadingExtent,
+  updateMapUI,
+} from '../../../modules/map/actions';
+import { getLeadingExtent } from '../../../modules/map/util';
+import {
+  getLayers,
+  getActiveLayers,
+} from '../../../modules/layers/selectors';
+import { getSelectedDate } from '../../../modules/date/selectors';
+import util from '../../../util/util';
+import { fly } from '../../../map/util';
+import * as layerConstants from '../../../modules/layers/constants';
+import * as compareConstants from '../../../modules/compare/constants';
+import * as paletteConstants from '../../../modules/palettes/constants';
+import * as vectorStyleConstants from '../../../modules/vector-styles/constants';
+import { LOCATION_POP_ACTION } from '../../../redux-location-state-customs';
+import { EXIT_ANIMATION, STOP_ANIMATION } from '../../../modules/animation/constants';
+import { SET_SCREEN_INFO } from '../../../modules/screen-size/constants';
+
+const UpdateProjection = (props) => {
+  const {
+    action,
+    activeLayers,
+    compare,
+    compareMapUi,
+    compareMode,
+    config,
+    dateCompareState,
+    fitToLeadingExtent,
+    getGranuleOptions,
+    isMobile,
+    layerState,
+    map,
+    models,
+    preloadForCompareMode,
+    proj,
+    projectionTrigger,
+    updateExtent,
+    updateLayerVisibilities,
+    updateMapUI,
```

Comment:
"updateMapUI" already declared on line 11.

Reply:
This is a Redux action using the same name.